modules/ROOT/pages/creating-a-custom-policy.adoc

```diff
@@ -0,0 +1,292 @@
+
+* Category
++
+Category of which the policy belongs. Used to group and filter policies in API Manager UI, any string value is valid. For example, security, quality of service, compliance, troubleshooting, and transformation.
+
+You can view this xref:https://github.com/mulesoft/ang-containers/blob/master/data/management/dialects/policy-definition.yaml[policy definition file] in GitHub with more information about all the supported fields. Some of the fields are auto-completed by the platform during the publication process, so not every field is read or is necessary in the file.
+
+The following code is an example of a basic authentication definition YAML file:
+----
+##%Policy Definition 0.1
+name: Basic authentication - Simple
+description: Enforces HTTP Basic authentication according to the details configured in the policy.
+category: Security
+providedCharacteristics:
+  - Requires authentication
+requiredCharacteristics: []
+interfaceScope: ["api",
+----
+
+[[create-implementation-files]]
+== Create Implementation Files
+The policy implementation files that are required to make the policy available in a runtime are a binary file and a metadata file. The binary file contains the policy logic and manages the runtime execution. The file can be a WebAssembly (WASM) for Anypoint Flex Gateway or a JAR file for Mule 4 runtime. The metadata file describes what technology the binary file applies to and from which version.
+
+=== Create Metadata YAML File
+The metadata YAML file provides details about the specific implementation of the policy definition. A single policy definition may several implementations, each an independent asset in Exchange or different versions of the same Exchange asset. 
+
+The metada YAML file must include the following fields:
+
+* Header
++
+#%Policy Implementation 1.0
+* Technology
++
+Either `mule4`` or `flexGateway`.
+* Name
+* minRuntimeVersion
++
+A semver version representing the first runtime version that the implementation binary is compatible with. For example, a mule binary might only be compatible from 4.3.0 onwards.
+
+The YAML file can also include:
+
+* Release notes
+* maxRuntimeVersion
++
+A semver version or range (only supporting missing patch/minor or using the `x` or `*` wildcards). For example, 1.2.3, 2.x, and 1.x.x.
+
+The following code is an example of a implementation metadata YAML file:
+
+----
+#%Policy Implementation 1.0
+minRuntimeVersion: 4.1.1
+maxRuntimeVersion: 4.4.0
+technology: mule4
+name: Simple Auth Mule Implementation
+releaseNotes: "<some notes here>"
+
+----
+
+=== Create the Implementation Binary Files
+The implementation binary file is a JAR file for Mule 4 policies or a WebAssembly WASM file for Anypoint Flex Gateway. These files support all of the configuration fields specified in the definition schema and function the way the definition metadata describes. Exchange validates the files to ensure that they are authentic and not a random file with a .wasm extension.
```

Comment:
"should support all of the configuration fields and function the way the definition metadata describes".

Please remove the last sentence regarding Exchange validation

Reply:
fixed